light_controller/ir_toy/ir_toy.py

```diff
@@ -0,0 +1,352 @@
+
+    def enableTX_ByteCount(self):
+        """
+        If enabled, the IR Toy returns the number of bytes transmitted at the end of a 0x03 command. The format is tXY,
+        where XY is the 16bit count of bytes transmitted:
+
+        't' is the character t
+        X is the high 8 bits of byte count
+        Y is the low 8 bits of byte count
+        :return: None
+        """
+        self.writeByte(CommandEnum.enableTX_ByteReport)
+
+    def enableTX_NotifyOnComplete(self):
+        """
+        If enabled, the IR Toy returns a success or error code at the end of a 0x03 transmit command.
+
+        Notify on complete codes
+        Code	Description
+        C		Success, no errors, complete
+        F		Buffer underrun during transmit
+        :return: None
+        """
+        self.writeByte(CommandEnum.enableTX_NotifyOnComplete)
+
+    def enableTX_Handshake(self):
+        """
+        If enabled, the IR Toy will request data packets during a transmit command. Each request is the number of bytes
+        free in the USB buffer, currently always 62. The software interfacing the IR Toy should wait for each request
+        before sending the data.
+        :return: None
+        """
+        self.writeByte(CommandEnum.enableTX_Handshake)
+
+    def reset(self):
+        """
+        Reset the device
+        :return: None
+        """
+        self.writeArray(self.TERMINATOR)
+        self.writeArray((CommandEnum.reset,) * 5)
+        time.sleep(0.05)
+
+    def getVersion(self):
+        """
+        Get the hardware and firmware versions
+        :return: hardware string, firmware version
+        """
+        try:
+            self.reset()
+            self.writeByte(ord('v'))
+            buffer = self.read(4)
+            hardware, revision = struct.unpack('>2s2s', buffer)
+            return hardware, int(revision)
+        finally:
+            self.setSamplingMode()
+
+    def setTransmitMode(self):
+        """
+        Setup the device for transmitting.
+        Enable handshakes and reports
+        :return: TX buffer size
+        """
+        self.enableTX_Handshake()
+        self.enableTX_ByteCount()
+        self.enableTX_NotifyOnComplete()
+        bufferSize = self.enableTransmitMode()
+        return bufferSize
+
+    def setSamplingMode(self):
+        """
+        Setup the device for sampling
+        :return: None
+        """
+        self.reset()
+        self.writeByte(ord('S'))
+        self.protocolVersion = self.read(3)
+
+    def receiveSignal(self):
+        """
+        Record a signal
+
+        :return: List of bytes representing the ir code.
+        """
+        self.setSamplingMode()
+        bytesToRead = 1
+        readCount = 0
+        irCode = []
+
+        while (True):
+            readByte = self.read(bytesToRead)
+            if not readByte:
+                continue
+            readByte = ord(readByte)
+            irCode.append(readByte)
+            if irCode[-2:] == self.TERMINATOR:
```

Comment:
Is there example data to show in the comments? Im not certain why the last 2 elements are used.